## Desafio-Ifood


In [34]:
import pandas as pd
import numpy as np

In [35]:
df = pd.read_csv("C:\\Users\\João Luís\\Desktop\\Desafio-Ifood\\data\\mkt_data.csv")

In [36]:
display(df)

,Unnamed: 0,Income,Kidhome,Teenhome,Recency,MntWines,MntFruits,MntMeatProducts,MntFishProducts,MntSweetProducts,...,education_Graduation,education_Master,education_PhD,MntTotal,MntRegularProds,AcceptedCmpOverall,marital_status,education_level,kids,expenses
0,0,58138.0,0,0,58,635,88,546,172,88,...,3.0,NaN,NaN,1529,1441,0,Single,Graduation,0,1529
1,1,46344.0,1,1,38,11,1,6,2,1,...,3.0,NaN,NaN,21,15,0,Single,Graduation,2,21
2,2,71613.0,0,0,26,426,49,127,111,21,...,3.0,NaN,NaN,734,692,0,Together,Graduation,0,734
3,3,26646.0,1,0,26,11,4,20,10,3,...,3.0,NaN,NaN,48,43,0,Together,Graduation,1,48
4,4,58293.0,1,0,94,173,43,118,46,27,...,NaN,NaN,5.0,407,392,0,Married,PhD,1,407
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
2200,2200,61223.0,0,1,46,709,43,182,42,118,...,3.0,NaN,NaN,1094,847,0,Married,Graduation,1,1094
2201,2201,64014.0,2,1,56,406,0,30,0,0,...,NaN,NaN,5.0,436,428,1,Together,PhD,3,436
2202,2202,56981.0,0,0,91,908,48,217,32,12,...,3.0,NaN,NaN,1217,1193,1,Divorced,Graduation,0,1217
2203,2203,69245.0,0,1,8,428,30,214,80,30,...,NaN,4.0,NaN,782,721,0,Together,Master,1,782


In [37]:
display(df.info())
display(df.shape)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2205 entries, 0 to 2204
Data columns (total 44 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            2205 non-null   int64  
 1   Income                2205 non-null   float64
 2   Kidhome               2205 non-null   int64  
 3   Teenhome              2205 non-null   int64  
 4   Recency               2205 non-null   int64  
 5   MntWines              2205 non-null   int64  
 6   MntFruits             2205 non-null   int64  
 7   MntMeatProducts       2205 non-null   int64  
 8   MntFishProducts       2205 non-null   int64  
 9   MntSweetProducts      2205 non-null   int64  
 10  MntGoldProds          2205 non-null   int64  
 11  NumDealsPurchases     2205 non-null   int64  
 12  NumWebPurchases       2205 non-null   int64  
 13  NumCatalogPurchases   2205 non-null   int64  
 14  NumStorePurchases     2205 non-null   int64  
 15  NumWebVisitsMonth    

None

(2205, 44)

Temos 2205 linhas e 44 colunas          
Temos 11 colunas float, 31 int (42 numericas) e 2objetc

In [38]:
display(df.duplicated().sum())

np.int64(0)

In [39]:
display(df.isnull().sum())

Unnamed: 0                 0
Income                     0
Kidhome                    0
Teenhome                   0
Recency                    0
MntWines                   0
MntFruits                  0
MntMeatProducts            0
MntFishProducts            0
MntSweetProducts           0
MntGoldProds               0
NumDealsPurchases          0
NumWebPurchases            0
NumCatalogPurchases        0
NumStorePurchases          0
NumWebVisitsMonth          0
AcceptedCmp3               0
AcceptedCmp4               0
AcceptedCmp5               0
AcceptedCmp1               0
AcceptedCmp2               0
Complain                   0
Z_CostContact              0
Z_Revenue                  0
Response                   0
Age                        0
Customer_Days              0
marital_Divorced        1975
marital_Married         1351
marital_Single          1728
marital_Together        1637
marital_Widow           2129
education_2n Cycle      2007
education_Basic         2151
education_Grad

Temos dados faltantes, porém como temos as colunas, martial_status e education_level completas, podemos ignorar essas colunas que tem valores nulos, ou simplesmente excluir essas colunas para limpar nosso df

In [40]:
df_limpo = df.drop(columns=["marital_Divorced", "marital_Married", "marital_Single", "marital_Together", 
                            "marital_Widow", "education_2n Cycle", "education_Basic", "education_Graduation", 
                            "education_Master", "education_PhD"])

In [41]:
display(df_limpo.describe())

,Unnamed: 0,Income,Kidhome,Teenhome,Recency,MntWines,MntFruits,MntMeatProducts,MntFishProducts,MntSweetProducts,...,Z_CostContact,Z_Revenue,Response,Age,Customer_Days,MntTotal,MntRegularProds,AcceptedCmpOverall,kids,expenses
count,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,2205.000000,...,2205.0,2205.0,2205.00000,2205.000000,2205.000000,2205.000000,2205.000000,2205.00000,2205.000000,2205.000000
mean,1102.000000,51622.094785,0.442177,0.506576,49.009070,306.164626,26.403175,165.312018,37.756463,27.128345,...,3.0,11.0,0.15102,51.095692,2512.718367,562.764626,518.707483,0.29932,0.948753,562.764626
std,636.672993,20713.063826,0.537132,0.544380,28.932111,337.493839,39.784484,217.784507,54.824635,41.130468,...,0.0,0.0,0.35815,11.705801,202.563647,575.936911,553.847248,0.68044,0.749231,575.936911
min,0.000000,1730.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,3.0,11.0,0.00000,24.000000,2159.000000,4.000000,-283.000000,0.00000,0.000000,4.000000
25%,551.000000,35196.000000,0.000000,0.000000,24.000000,24.000000,2.000000,16.000000,3.000000,1.000000,...,3.0,11.0,0.00000,43.000000,2339.000000,56.000000,42.000000,0.00000,0.000000,56.000000
50%,1102.000000,51287.000000,0.000000,0.000000,49.000000,178.000000,8.000000,68.000000,12.000000,8.000000,...,3.0,11.0,0.00000,50.000000,2515.000000,343.000000,288.000000,0.00000,1.000000,343.000000
75%,1653.000000,68281.000000,1.000000,1.000000,74.000000,507.000000,33.000000,232.000000,50.000000,34.000000,...,3.0,11.0,0.00000,61.000000,2688.000000,964.000000,884.000000,0.00000,1.000000,964.000000
max,2204.000000,113734.000000,2.000000,2.000000,99.000000,1493.000000,199.000000,1725.000000,259.000000,262.000000,...,3.0,11.0,1.00000,80.000000,2858.000000,2491.000000,2458.000000,4.00000,3.000000,2491.000000


In [42]:
desvio = df_limpo["Income"].std()
media = df_limpo["Income"].mean()

print((desvio/media)*100)

40.12441554786958


A distribuição de renda esta muito diferente, o valor máximo é de 113.734.0 e o menor é de 1730.0, puxando a média la pra cima e deixando o desvio padrão em 40,12%

In [45]:
df_limpo["education_level"].value_counts()

education_level
Graduation    1113
PhD            476
Master         364
2n Cycle       198
Basic           54
Name: count, dtype: int64

o nivel predominante é de Graduation com 1113 ocupantes e o menor é o Basic com 54

In [44]:
df_limpo["marital_status"].value_counts()

marital_status
Married     854
Together    568
Single      477
Divorced    230
Widow        76
Name: count, dtype: int64

In [53]:
df_marital_kids = df_limpo.groupby("marital_status")["kids"].mean()
display(df_marital_kids)

marital_status
Divorced    1.008696
Married     0.967213
Single      0.859539
Together    0.980634
Widow       0.881579
Name: kids, dtype: float64

As médias de filhos comparados ao estado civil, são muito próximas, então nao podemos dizer que pessoas casadas tendem a ter mais filhos

In [57]:
df_expenses_kids = df_limpo.groupby("kids")["expenses"].mean()
display(df_expenses_kids)

kids
0    1041.207006
1     434.530576
2     221.568675
3     237.380000
Name: expenses, dtype: float64

As pessos, tendem a gastar menos quando se tem filhos e os valores vão abaixando conforme a quantidade de filhos


In [59]:
correlacao = df_limpo["Income"].corr(df_limpo["expenses"])
print(correlacao)

0.8230660021398378


A correlação entre sálario e gastas com delivery mostra que quem ganha mais, costuma gastar mais

Com Base nas Analises acima, percebemos que quem gasta menos com delivery, são as pessoas que tem os menores sálarios e também pessoas que possuem filho(os), dessa forma uma maneira
de fazer eles comprarem mais seria a disponibilizaram de cupons de descontos, ofertas de combos e até mesmo frete grátis ou com valores promocionais tanto nas compras quantos nos produtos.

## Problema de Negócios: Análise de Conversão e "Vitrinismo" digital
Precisamos entender se o alto tráfego na nossa plataforma (visitas no app/site) está realmente se convertendo em vendas frequentes, ou se temos um gargalo onde os clientes entram, olham os restaurantes, mas saem sem pedir nada.

Perguntas necessárias para responder a esse problema:

Qual é a correlação entre o número de visitas no mês (NumWebVisitsMonth) e o tempo desde a última compra (Recency)? Pessoas que visitam mais compram mais rápido?

In [60]:
correlacao_WebVisit_Recency = df_limpo["NumWebVisitsMonth"].corr(df_limpo["Recency"])
print(correlacao_WebVisit_Recency)

-0.017905758576814878


não existe correlação com o número de visitas e o tempo que o cliente fica sem comprar algo, porém existe um problema na demora para escolher o pedido, seria interessante cupons flash, onde o cliente abre o aplicativo e ja tem um cupom para fazer a compra